newIDE/app/src/ObjectsList/index.js

```diff
@@ -296,17 +396,53 @@ const ObjectsList = React.forwardRef<Props, ObjectsListInterface>(
       [forceUpdate, forceUpdateList, unsavedChanges]
     );
 
-    const deleteObject = React.useCallback(
-      (i18n: I18nType, objectWithContext: ObjectWithContext) => {
-        const { object, global } = objectWithContext;
-
-        const answer = Window.showConfirmDialog(
-          i18n._(
-            t`Are you sure you want to remove this object? This can't be undone.`
-          )
+    const selectObjectFolderOrObjectWithContext = React.useCallback(
+      (objectFolderOrObjectWithContext: ?ObjectFolderOrObjectWithContext) => {
+        onObjectFolderOrObjectWithContextSelected(
+          objectFolderOrObjectWithContext
         );
+      },
+      [onObjectFolderOrObjectWithContextSelected]
+    );
+
+    const deleteObjectFolderOrObjectWithContext = React.useCallback(
+      async (
+        objectFolderOrObjectWithContext: ?ObjectFolderOrObjectWithContext
+      ) => {
+        if (!objectFolderOrObjectWithContext) return;
+        const {
+          objectFolderOrObject,
+          global,
+        } = objectFolderOrObjectWithContext;
+        if (objectFolderOrObject.isFolder()) {
+          if (objectFolderOrObject.getChildrenCount() === 0) {
+            objectFolderOrObject
+              .getParent()
+              .removeFolderChild(objectFolderOrObject);
+            forceUpdateList();
+            selectObjectFolderOrObjectWithContext(null);
+          }
+          return;
+        }
+
+        const answer = await showDeleteConfirmation({
+          message: t`Are you sure you want to remove this object? This can't be undone.`,
+          title: t`Remove object`,
+        });
```

Comment:
just to check, this does not require typing the name of the object to delete it right?

Reply:
Indeed!